src/ppscore/calculation.py

```diff
@@ -52,24 +54,33 @@ def _calculate_model_cv_score_(df, target, feature, metric, model, **kwargs):
         target_series = df[target]
 
     # preprocess feature
+    preprocess = None
     if df[feature].dtype == object:
-        one_hot_encoder = preprocessing.OneHotEncoder()
-        sparse_matrix = one_hot_encoder.fit_transform(df[feature].values.reshape(-1, 1))
-        feature_df = sparse_matrix
+        # Dealing with categorical feature here here:
+        preprocess = ColumnTransformer(transformers = [('ct',OneHotEncoder(),[feature])])
+
+    # reshaping needed because there is only 1 feature: coerce to DataFrame
+    feature_df = df[feature].to_frame()
+
+    if preprocess is None:
+        pipeline_model = model
     else:
-        # reshaping needed because there is only 1 feature
-        feature_df = df[feature].values.reshape(-1, 1)
+        pipeline_model = make_pipeline(preprocess, model)
+
+    # Pull the groups out if passed
+    groups = kwargs.get('groups', None)
 
-    # Crossvalidation is stratifiedKFold for classification, KFold for regression
```

Comment:
It would be great to keep the annotation somewhere when no explicit CV object is passed but just the number of folds

Reply:
it is already in the `score` function. I also added it to the docstring